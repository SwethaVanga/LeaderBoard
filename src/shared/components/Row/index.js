import React from 'react'
import style from './index.module.scss'
import { ReactSVG } from 'react-svg'
import edit from '../../../assets/images/edit.png'
import classNames from 'classnames'
import Arrow from '../../../assets/images/Arrow.svg'
import { getUserAvatar } from '../../../utils/helpers'

const Row = ({
  position,
  changePosition,
  showModalUpdate,
  editUser,
  player,
}) => {
  const cx = classNames.bind(style)
  return (
    <div className={style.row}>
      <div className={style.row__inner}>
        <div className={style.row__index}>
          {position === 1
            ? `${position}st`
            : position === 2
              ? `${position}nd`
              : position === 3
                ? `${position}rd`
                : `${position}th`}
        </div>
        <div className={style.row__image}>{getUserAvatar(player.name)}</div>
        <div className={style.row__score}>{player.score}</div>
        <div className={style.row__name}>{player.name}</div>
      </div>
      <div className={style.row__inner}>
        <div
          className={cx(style.row__change, {
            [style.change__up]: changePosition > 0,
            [style.change__down]: changePosition < 0,
            [style.change__zero]: changePosition === 0,
          })}
        >
          <ReactSVG src={Arrow} />
          {changePosition === 0
            ? 'No change'
            : Math.abs(changePosition) === 1
              ? '1 place'
              : `${Math.abs(changePosition)} places`}
        </div>
        <img
          src={edit}
          className={style.row__edit}
          onClick={() => {
            showModalUpdate()
            editUser(position - 1)
          }}
          alt="edit"
        />
      </div>
    </div>
  )
}

export default Row
