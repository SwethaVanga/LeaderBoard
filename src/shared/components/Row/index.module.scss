@import "../../../assets/style/variables.scss";
.row {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 38px;
  margin-bottom: 29px;

  &__inner {
    display: flex;
    align-items: center;
    height: 34px;
  }

  &__index {
    width: 40px;
    margin-right: 23px;
    font-family: "Gilroy";
    font-style: normal;
    font-weight: bold;
    font-size: 20px;
    line-height: 21px;
    color: $index;
  }

  &__image {
    width: 38px;
    height: 38px;
    margin-right: 42px;
    & img {
      border-radius: 10px;
    }
  }

  &__score {
    width: 40px;
    margin-right: 37px;
    font-family: "Gilroy";
    font-style: normal;
    font-weight: bold;
    font-size: 30px;
    line-height: 21px;
    color: $table-score;
  }

  &__name {
    font-family: Lato;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 21px;
    color: $title;
  }

  &__edit {
    margin-left: 34px;
    cursor: pointer;
  }

  &__change {
    display: flex;
    align-items: center;
    position: relative;
    font-family: Heebo;
    font-style: normal;
    font-weight: normal;
    font-size: 15px;
    line-height: 22px;

    & svg {
      padding-right: 12px;
    }
  }
}
.change {
  &__zero {
    color: $oval;

    & path {
      fill: $oval;
    }
  }

  &__up {
    color: $score;

    & svg {
      transform: rotate(-90deg);
    }
    & path {
      fill: $score;
    }
  }

  &__down {
    color: $red;

    & svg {
      transform: rotate(90deg);
    }
    & path {
      fill: $red;
    }
  }
}
