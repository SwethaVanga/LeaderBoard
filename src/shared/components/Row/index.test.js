import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import store from '../../../redux/root.js'
import Row from '../Form/index.js'

it('render correctly row component', () => {
  const props = {
    position: 1,
    changePosition: 1,
    player: {
      name: '',
      score: 1,
    },
    showModalUpdate: () => false,
    editUser: () => 1,
  }
  const RowComponent = renderer
    .create(
      <Provider store={store}>
        <Row
          position={props.position}
          changePosition={props.changePosition}
          player={props.player}
          showModalUpdate={props.showModalUpdate}
          editUser={props.editUser}
        />
      </Provider>
    )
    .toJSON()
  expect(RowComponent).toMatchSnapshot()
})
