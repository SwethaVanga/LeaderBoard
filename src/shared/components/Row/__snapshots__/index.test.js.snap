// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly row component 1`] = `
<div
  className="form"
>
  <div
    className="form__title"
  >
    Add user score
  </div>
  <div
    className="form__inner"
  />
</div>
`;
